src/compas_timber/ghpython/components/CT_Model/code.py

```diff
@@ -60,8 +60,11 @@ def RunScript(self, Elements, JointRules, Features, MaxDistance, CreateGeometry)
         if Features:
             features = [f for f in Features if f is not None]
             for f_def in features:
-                for element in f_def.elements:
-                    element.add_features(f_def.feature)
+                if not f_def.elements:
+                    self.AddRuntimeMessage(Warning, "Features defined in model must have elements defined. Features without elements will be ignored")
+                elif f_def.geometries:
```

Comment:
`else` here is enough
actually you can drop it altogether. the `for` will not enter if there's no elements